test/api/search_controller_test.rb

```diff
@@ -245,4 +245,38 @@ class SearchControllerTest < ActionDispatch::IntegrationTest
 
     puts "#{File.basename(__FILE__)}: #{self.method_name} successful!"
   end
+
+  test 'should run inferred search using facets' do
+    puts "#{File.basename(__FILE__)}: #{self.method_name}"
+
+    convention_study = Study.find_by(name: "Test Study #{@random_seed}")
+    other_study = Study.find_by(name: "API Test Study #{@random_seed}")
+    original_description = other_study.description.to_s.dup
+    species_facet = SearchFacet.find_by(identifier: 'species')
+    facet_query = "#{species_facet.identifier}:#{species_facet.filters.first[:id]}"
+    execute_http_request(:get, api_v1_search_path(type: 'study', facets: facet_query))
+    assert_response :success
+    expected_accessions = [convention_study.accession]
+    assert_equal expected_accessions, json['matching_accessions'],
+                 "Did not find expected accessions before inferred search, expected #{expected_accessions} but found #{json['matching_accessions']}"
+
+    # now update non-convention study to include a filter display value in its description
+    # this should be picked up by the "inferred" search
+    filter_name = species_facet.filters.first[:name]
+    other_study.update(description: filter_name)
+    facet_query = "#{species_facet.identifier}:#{species_facet.filters.first[:id]}"
+    execute_http_request(:get, api_v1_search_path(type: 'study', facets: facet_query))
+    assert_response :success
+    inferred_accessions = [convention_study.accession, other_study.accession]
+    assert_equal inferred_accessions, json['matching_accessions'],
+                 "Did not find expected accessions after inferred search, expected #{inferred_accessions} but found #{json['matching_accessions']}"
+    inferred_study = json['studies'].last # inferred matches should be at the end
+    assert inferred_study['inferred_match'],
+           "Did not mark last search results as inferred_match: #{inferred_study['inferred_match']} != true"
+
+    # reset description so other tests aren't broken
+    other_study.update(description: original_description)
+
+    puts "#{File.basename(__FILE__)}: #{self.method_name} successful!"
+  end
```

Comment:
This is good.  Can you also add a test for 
1. Searching for multiple facets/filters behaves as expected
2. Searching for an exact phrase + facet behaves as expected.  
I expect both of these cases to work, but given that this is home-page behavior, an extra layer of testing would be good

Reply:
Can do.

Reply:
To be clear, we already have a test for the first case.  The existing test `'should return search results using facets'` uses multiple facets at the same time.  But I can add a test for the second case.

Reply:
70f04a9